### PR TITLE
CSR-346 Created DS Eligibility

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Scheme.php [hunk 1]
```diff
@@ -47,8 +54,8 @@ public function getName()
     }
 
     /**
-     * @param string $name
-     * @return Scheme
+     * @param $name
+     * @return $this
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Scheme.php]: `@return Scheme`

I fixed this ones :wink:

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        }
+        $this->hasDegreeRequirement = $hasRequirement;
+
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    public function getHasDegreeRequirement()
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: Should be `hasDegreeRequirement`

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        }
+        $this->isBritishCitizen = $hasRequirement;
+
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    public function getIsBritishCitizen()
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: `getIs...` doesn't make sense

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        return $this->hasResidenceRequirement;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isEligible()
+    {
+        return (
+            $this->hasDegreeRequirement &&
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: Personally, I would renamed the properties to be `degreeRequirement`, then have the getter be `hasDegreeRequirement`, and use them here

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param $hasRequirement
+     * @return $this
+     */
+    public function setHasDegreeRequirement($hasRequirement)
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: `setHas..`?

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    public function hasDegreeRequirement()
+    {
+        return $this->degreeRequirement;
+    }
+
+    /**
+     * @param $hasRequirement
+     * @return $this
+     */
+    public function setIsBritishCitizen($hasRequirement)
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: `setIs...`, should it not just be `setBritishCitizen`?

**Comment by eddiejaoude** [comment]
Build is broken with... (looks like param needs updating in PHPSpec Test)

```php
TransformCore/Bundle/CsrFastStreamBundle/Form/SchemeEligibility/Diplom
aticServiceEligibilityFormType                                        
  25  - it should build form
      method call:
        Double\FormBuilderInterface\P63->add("degreeRequirement", "checkbox", ["label" => "diplomatic-service.selection.degree", "required" => false])
      was not expected.
      Expected calls are:
        - add(exact("hasDegreeRequirement"), exact("checkbox"), exact(["label" => "diplomatic-service.selection.degree", "required" => false]))
        - add(exact("isBritishCitizen"), exact("checkbox"), exact(["label" => "diplomatic-service.selection.citizen", "required" => false]))
        - add(exact("hasResidenceRequirement"), exact("checkbox"), exact(["label" => "diplomatic-service.selection.residence", "required" => false]))
        - add(exact("update"), exact("submit"), exact(["scheme.selection.form.updateButton.label"]))
```

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+     */
+    protected $britishCitizen = false;
+
+    /**
+     * @var bool
+     */
+    protected $residenceRequirement = false;
+
+    /**
+     * @param $id
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: `int`

I will update it though

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+     * @return int
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param $hasRequirement
+     * @return $this
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: `@return DiplomaticServiceEligibility`

I will update

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+     * @return bool
+     */
+    public function hasDegreeRequirement()
+    {
+        return $this->degreeRequirement;
+    }
+
+    /**
+     * @param $hasRequirement
+     * @return $this
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: `@return DiplomaticServiceEligibility`

I will update

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+     * @return bool
+     */
+    public function isBritishCitizen()
+    {
+        return $this->britishCitizen;
+    }
+
+    /**
+     * @param $hasRequirement
+     * @return $this
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: `@return DiplomaticServiceEligibility`

I will update

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+    public function isBritishCitizen()
+    {
+        return $this->britishCitizen;
+    }
+
+    /**
+     * @param bool $hasRequirement
+     * @return DiplomaticServiceEligibility
+     */
+    public function setResidenceRequirement($hasRequirement)
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: `$hasRequirement` this **variable** doesn't make sense here

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    public function hasDegreeRequirement()
+    {
+        return $this->degreeRequirement;
+    }
+
+    /**
+     * @param bool $hasRequirement
+     * @return DiplomaticServiceEligibility
+     */
+    public function setBritishCitizen($hasRequirement)
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/DiplomaticServiceEligibility.php]: `$hasRequirement` this **variable** doesn't make sense here

**Comment by eddiejaoude** [comment]
Can you add the Doctrine config (yml)? Then I can generate a diagram

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Scheme.php [hunk 13]
```diff
@@ -2,6 +2,8 @@
 
 namespace TransformCore\Bundle\CsrFastStreamBundle\Entity;
 
+use TransformCore\Bundle\CsrFastStreamBundle\Entity\SchemeEligibility\SchemeEligibility;
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Scheme.php]: Namespace looks incorrect?

**Comment by eddiejaoude** [comment]
@tomrup A Scheme belongs to a Programme?

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/SchemeEligibility.php [hunk 14]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+
+namespace TransformCore\Bundle\CsrFastStreamBundle\Entity\SchemeEligibility;
+
+/**
+ * Class SchemeEligibility
+ * @package TransformCore\Bundle\CsrFastStreamBundle\Entity\SchemeEligibility
+ */
+interface SchemeEligibility
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/SchemeEligibility.php]: Do you plan to use a **Discriminator Map** for this?

Comment (tomrup) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/SchemeEligibility.php]: Possibly the approach for Beta - noyt needed for alpha I think.

Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/SchemeEligibility.php]: For the **Discriminator Map**?

**Comment by eddiejaoude** [comment]
PR into this branch https://github.com/TransformCore/CSR-Fast-Stream-Domain-Model/pull/43

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Scheme.php (hunk 13 above)
Comment (brettminnie) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Scheme.php]: Use statement on interface :)

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/SchemeEligibility.php (hunk 14 above)
Comment (brettminnie) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/SchemeEligibility/SchemeEligibility.php]: Eventually yes

**Pull request merged**